### DISCUSSION TITLE
z/OS Open Source Guild Meeting #2 - September 21, 2022 at 11am EST

### DISCUSSION THREAD

**Discussion by IgorTodorovskiIBM** [zopencommunity/meta#36] [Guild]
### Agenda

- Highlights in z/OS Open Source
- How to obtain and use the latest z/OS Open Source ports
- Deep dive into the z/OS Open Source porting process
- Discussion on expected behaviour of z/OS Open Source tools
- Polls and Open Discussion

### How to Join?
Join link: https://ibm.webex.com/ibm/j.php?MTID=m72857a0933fbcd1c58c116a13c9f1e7b 
Webinar number: 145 105 4264
Webinar password: QamxS8QX2m7 (72697879 from phones)

### Date and Time
* Sept. 21, 11am-12pm EST (8am PST)

### Recording
[Recording link](http://128.168.139.253:8081/collateral/Guild%20Meetings/Sept%2021%202022/Guild_Recording.mp4)

### Powerpoint Slides
[Open Source Guild September 2022.pptx](https://github.com/ZOSOpenTools/meta/files/9618608/Open.Source.Guild.September.2022.pptx)

[Guild 2 survey results](https://github.com/ZOSOpenTools/meta/files/9618610/guild2_survey.pdf)

**Comment by IgorTodorovskiIBM** [comment]
Topic considerations: 
* Open source contribution process
* Deep dive into a port (from environment setup, to porting process, to contribution upstream)
* Overview of common gotchas
* CI/CD pipeline

These are mostly taken from the survey results from our first Guild meeting. Please let us know if you have other topics in mind!

**Reply by IgorTodorovskiIBM** [reply]
Another potential topic:
* Discussion of a common library for porting - which implements missing POSIX functions, overrides certain LE functions to match Linux, and provides consistency in how ASCII/EBCDIC is handled

**Reply by MikeFultonDev** [reply]
@IgorTodorovskiIBM this meeting is titled 'initial meeting' but it's the second one - can you change the title?

**Reply by MikeFultonDev** [reply]
ah. i was able to change it...

**Reply by IgorTodorovskiIBM** [reply]
Thanks!

**Reply by IgorTodorovskiIBM** [reply]
* Mailing list or Github discussions

**Comment by IgorTodorovskiIBM** [comment]
If you are interested in an Asian-Pacific Call, let us know by upvoting this comment!

**Reply by MikeFultonDev** [reply]
Another possible topic for discussion: How to 'deep dive' for a particular software package? I would like to hear from folks about how to enhance 'git' for z/OS so that it can work really well for both traditional EBCDIC source packages (e.g. a COBOL CICS application) as well as fully-UTF8 applications (e.g. any of the github repos we have under zosopentools). There are a few different knobs:
- 'git configure'
- .gitattributes
- environment variables
- command-line options
- others?

**Comment by dboyes99** [comment]
git server setup for z/os and z/VM would be an interesting presentation. Also use of tools like gitea.

**Reply by IgorTodorovskiIBM** [reply]
Thank you! By Git server, are you referring to something along these lines? https://www.inmotionhosting.com/support/website/git/git-server/

Git is on our radar. We're actually working towards the goal of getting Git ported to z/OS and upstreamed. But first, we're in the process of porting some of its dependencies (including but not limited to: make m4 perl autoconf automake help2man makeinfo xz zlib openssl curl expat gettext).

**Comment by dboyes99** [comment]
That URL is the really basic part, but something a little easier to use would probably grease the wheels a bit.
  Gmake and perl are available for z/OS from Rocket Software.

You may want to look closer at https://gitea.io/en-us/ as a base to move forward from. It would be interesting to see how difficult translating man and info
files to SCRIPT source would be. I suspect if the Bookmaster macros or the DCF SGML subset are available, the transition would be pretty simple to get really
nice looking output, but gtroff would probably do as well.

From: Igor Todorovski ***@***.***>
Reply-To: ZOSOpenTools/meta ***@***.***>
Date: Tuesday, September 20, 2022 at 12:21 PM
To: ZOSOpenTools/meta ***@***.***>
Cc: dboyes99 ***@***.***>, Comment ***@***.***>
Subject: Re: [ZOSOpenTools/meta] z/OS Open Source Guild Meeting #2 - September 21, 2022 at 11am EST (Discussion #36)


Thank you! By Git server, are you referring to something along these lines? https://www.inmotionhosting.com/support/website/git/git-server/

**Comment by KipTwitchell** [comment]
I'd like to join the next meeting, but couldn't see a schedule for when it will be held?

**Reply by IgorTodorovskiIBM** [reply]
Hi Kip, send me an email (itodorov@ca.ibm.com) with your name/email and I will invite to our monthly calls.  I'll post the recording for today's meeting tomorrow.

**Comment by IgorTodorovskiIBM** [comment]
Slides and survey now available:
[Open Source Guild September 2022.pptx](https://github.com/ZOSOpenTools/meta/files/9618608/Open.Source.Guild.September.2022.pptx)

[Guild 2 survey results](https://github.com/ZOSOpenTools/meta/files/9618610/guild2_survey.pdf)

Recording will be available tomorrow

**Comment by IgorTodorovskiIBM** [comment]
Here's the link to the recording: [link](http://128.168.139.253:8081/collateral/Guild%20Meetings/Sept%2021%202022/Guild_Recording.mp4)

**Reply by jrnewton** [reply]
@IgorTodorovskiIBM - recording link is broken. It's using a hardcoded (private) IP...

**Reply by IgorTodorovskiIBM** [reply]
Thanks, it should be back up. @MikeFultonDev, we may want to consider using a cloud service to store the recordings in. Any suggestions?
